ETC Version release requested changes implemented

sanchitsabhlok opened the pull request:
# Description
Made changes requested to comply with Release guide

## Changes Made
Modified README and changelog doc with the new repo name, cleaned language on Changelog, added credit for the original fork and date of fork to changelog, modified name in tests and notebooks.

sanchitsabhlok committed:
Created Version Release

Created an initial release version by starting with the template and updating to add files from the ETC.

sanchitsabhlok committed:
Removed pre commit hook

sanchitsabhlok committed:
Added Changelog comparison between the last stable UASAL Exposure Time Calculator and ETC 1.0

sanchitsabhlok committed:
Updated for Release

Modified README, tests, changelog and added relevant information regarding original repo. Renamed to stp_etc_imaging

sanchitsabhlok committed:
Updated filter addition

Updated the filter to be added using from_file instead of from_filter

sanchitsabhlok committed:
Finalized ETC changes for release

Updated changelog, fixed notebook

sanchitsabhlok committed:
Removed ci.yaml for now

sanchitsabhlok committed:
Addressed PR comments

Fixed tests with incorrect try/except implementation, fixed ETC changelog, removed _version file and fixed import module checks.

sanchitsabhlok committed:
Separated ETC installation and Environment checks into a bin script

sanchitsabhlok committed:
Added test tomls

sanchitsabhlok committed:
ETC now accepts test tomls

sanchitsabhlok committed:
Tests modified to check against test and real configs

sanchitsabhlok committed:
ETC now accepts custom toml directories and custom support data directories

sanchitsabhlok committed:
Removed get_data_path as this is now part of utils_config

sanchitsabhlok committed:
Exceptions for UASAL package imports added

sanchitsabhlok committed:
Clarified Text

Added telescope/instrument sections, paper links, minor changes and UM WCC support. Added language to clarify package dependency.

patrickingraham reviewed (commented):
A few minor comments. The major remaining item is the dependencies (which already has an open thread, but I didn't add any additional comment there)

sanchitsabhlok committed:
Minor changes.

Fixed typos, fixed imports for configs. Removed if name main clause in tests and fixed return call in config checks.

sanchitsabhlok committed:
Fixed pyproject.toml

Updated python version to 3.12. Added dependencies and optional dependencies.

patrickingraham reviewed (approved):
One comment. Thanks!
Please make sure you squash before merging.

sanchitsabhlok committed:
Added warning to the failure message for sensor initialization test
